### PR TITLE
Updated how Garden updates work

### PR DESCRIPTION
**Pull request by TheBurchLog**
Moved the PATCH back to the API
Updated Routing Logic to only accept events from the local garden or the one who forwarded thee event. 
Removed the PATCH logic from garden.py and added config update logic.

### CODE REVIEW

File: src/app/beer_garden/api/http/handlers/v1/garden.py [hunk 1]
```diff
@@ -109,17 +111,38 @@ class GardenAPI(BaseHandler):
...
+                        args=[garden_name, "RUNNING"],
+                    )
+                )
+            elif operation == "config":
+                response = await self.client(
+                    Operation(
+                        operation_type="GARDEN_UPDATE_CONFIG",
+                        args=[
+                            garden_name,
+                            SchemaParser.parse_garden(op.payload, from_string=True),
```
Comment (hazmat345) [inline src/app/beer_garden/api/http/handlers/v1/garden.py]: Have you tested this? I don't believe a patch operation has a payload field.

File: src/app/beer_garden/garden.py [hunk 2]
```diff
@@ -35,30 +35,13 @@ def get_gardens() -> List[Garden]:
...
-            garden = update_garden_status(garden_name, operation.upper())
-        elif operation == "heartbeat":
-            garden = update_garden_status(garden_name, "RUNNING")
-
-        else:
-            raise ModelValidationError(f"Unsupported operation '{op.operation}'")
-
-    return garden
+    update_garden(db_garden)
+    return db_garden
```
Comment (hazmat345) [inline src/app/beer_garden/garden.py]: Every function in this module should be returning the result of the update:

```python
return update_garden(db_garden)
```

File: src/app/beer_garden/api/http/handlers/v1/garden.py (hunk 1 above)
Comment (TheBurchLog) [reply src/app/beer_garden/api/http/handlers/v1/garden.py]: You are right, it should have been `op.value`

File: src/app/beer_garden/garden.py (hunk 2 above)
Comment (TheBurchLog) [reply src/app/beer_garden/garden.py]: Fixed

**Pull request merged**